### PR TITLE
feat(input): adds autocomplete prop

### PR DESCRIPTION
**Pull request by chasestarr**
#### Scope

- [x] Minor: New Feature

### CODE REVIEW

File: src/input/types.js [hunk 1]
```diff
@@ -72,6 +72,8 @@ export type BaseInputPropsT<T> = {
   'aria-describedby': ?string,
   /** Defines styles for inputs that are grouped with other controls. */
   adjoined: AdjoinedT,
+  /** Determines if browser should provide value suggestions. */
+  autocomplete: boolean,
```
Comment (gergelyke) [inline src/input/types.js]: shouldn't it be a string value? https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete

Comment (chasestarr) [reply src/input/types.js]: thank you

File: src/input/base-input.js [hunk 2]
```diff
@@ -29,6 +29,7 @@ class BaseInput<T: EventTarget> extends React.Component<
     'aria-labelledby': null,
     'aria-describedby': null,
     adjoined: ADJOINED.none,
+    autocomplete: 'on',
```
Comment (tajo) [inline src/input/base-input.js]: Should we be consistent with `autoFocus` and use `autoComplete` ?

Comment (chasestarr) [reply src/input/base-input.js]: sure - I can update that

**Pull request merged**